packages/repository/src/model.ts

```diff
@@ -156,6 +156,16 @@ export class ModelDefinition {
     );
     return idProps;
   }
+
+  /**
+   * Returns the primary key(s) of a given model.
+   * Implemented by Raphael Drai for the purpose of PUT operations
+   * in relations models.
+   *
+   */
+  idNames(): string[] {
+    return this.idProperties();
```

Comment:
I am confused about the purpose of this method. Why cannot you call directly `this.idProperties` instead?

I am also arguing that `PUT` operation does not make much sense for relations and that we should remove it from this pull request. At which point `idNames` is not needed at all.

Reply:
isNames()  method should exist for the PUT operation due to the abstract class "export abstract class Entity" defined in packages/repository/src/model.ts that invokes this method in line 272, see below the copy/past.
But as per your proposing, I am removing it and it will be handled later when PUT will be defined. I will open a new PR for that purpose.

267   static getIdOf(entityOrData: AnyObject): any {
268      if (typeof entityOrData.getId === 'function') {
269        return entityOrData.getId();
270      }
271
272      const idName = this.definition.idName();
273      return entityOrData[idName];
274   }